### ISSUE TITLE
Lots of errors, both at startup and on the website itself.

### ISSUE THREAD

**Issue by smeagolthellama** [ryceg/Eigengrau-s-Essential-Establishment-Generator#482]
**Describe the bug**

When I open the website, The following two errors pop up:

```
Apologies! An error has occurred. You may be able to continue, but some parts may not work properly.

Error [tw-user-script-0]: expected expression, got '.'.
```
```
Apologies! An error has occurred. You may be able to continue, but some parts may not work properly.

Error [StoryInit]: <<run>>: bad evaluation: setup.urlSeed is not a function.
```

After the errors are `ok`-ed, the page is literally filled with errors.

![image](https://user-images.githubusercontent.com/20841523/103635408-25cbf180-4f51-11eb-86bb-9d0601ad9798.png)
 
**To Reproduce (bugs MUST be replicable)**

Town Seed (click [here](https://github.com/ryceg/Eigengrau-s-Essential-Establishment-Generator/wiki/FAQ#people-keep-talking-about-seeds-what-is-a-seed) for info): #idealisticsilvergrunion most recently, but pretty much any seed seems to trigger it.

Page to navigate to, buttons to press, etc: just the home page.

**Additional context**

I'm using Firefox 70.0b2 on Linux (devuan beowulf, based on debian), with a number of plugins installed:
 - noscript has all script sources set to allow
 - privacy badger didn't block anythin
 - ublock origin seems to be blocking some things from sentry.io, but the errors persist when ublock is turned off
 - pluckeye doesn't claim to have blocked anything, and it doesn't generally block anything without logging it.
 - I doubt any of the others did anything.

After clearing cookies and site data, the page looks like this:
![image](https://user-images.githubusercontent.com/20841523/103636463-af2ff380-4f52-11eb-9f1b-f9fa29ceeb20.png)

**Comment by smeagolthellama** [comment]
Extra information: the console had this in it after the second screenshot was taken:
```
Invalid URI. Load of media resource data: failed. eigengrausgenerator.com
Initializing sentry. sentry.ts:6:10
SyntaxError: "expected expression, got '.'"
    evalJavaScript (index):30047
    evalJavaScript (index):30047
    t (index):30051
    <anonymous> (index):30052
    e (index):57
    t (index):57
    i helpers.js:72
instrument.js:109:45
Hello world! instrument.js:109:45
Error: <<run>>: bad evaluation: setup.urlSeed is not a function
	<<run setup.urlSeed()>> instrument.js:109:45
Error: <<run>>: bad evaluation: setup.init is not a function
	<<run setup.init()>> instrument.js:109:45
Error: <<set>>: bad evaluation: setup.createTownBiome is not a function
	<<set $tempTown to setup.createTownBiome()>> instrument.js:109:45
Error: <<set>>: bad evaluation: setup.createTown is not a function
	<<set $town to setup.createTown($tempTown)>> instrument.js:109:45
Error: "<<run>>: bad evaluation: setup.urlSeed is not a function"
    value (index):30047
    t (index):30050
    <anonymous> (index):30052
    e (index):57
    t (index):57
    i helpers.js:72
instrument.js:109:45
Error: macro <<liveblock>> does not exist
	<<liveblock>> instrument.js:109:45
Error: macro <</liveblock>> does not exist
	<</liveblock>> instrument.js:109:45
Error: macro <<liveblock>> does not exist
	<<liveblock>> instrument.js:109:45
Error: macro <</liveblock>> does not exist
	<</liveblock>> instrument.js:109:45
Error: <<print>>: bad evaluation: State.variables.town is undefined
	<<print $town.type.toUpperFirst()>> instrument.js:109:45
Error: <<print>>: bad evaluation: State.variables.town is undefined
	<<print lib.firstCharacter($town.name)>> instrument.js:109:45
Error: <<print>>: bad evaluation: State.variables.town is undefined
	<<print lib.createTippyFull("With a population of " + $town.population + ".", $town.type)>> instrument.js:109:45
Error: <span>: bad evaluation from attribute directive "@title": State.variables.town is undefined
	<span @title="lib.formatPercentile(lib.sortArray($town.demographicPercentile))" class="tip dotted">… instrument.js:109:45
Error: <<print>>: bad evaluation: State.variables.town is undefined
	<<print lib.getPredominantRace($town.demographicPercentile).amountDescriptive>> instrument.js:109:45
Error: <<if>>: bad conditional expression in <<if>> clause: State.variables.town is undefined
	<<if $town.ignoreGender is false>><<print lib.articles.find($town.equality)>> <<print lib.createTippyFull($town.equalityDescription, $town.equality)>> 
	\<<else>>
	\<<print lib.articles.find($town.economicIdeologyIST)>> <</if>> instrument.js:109:45
Error: <<print>>: bad evaluation: setup.politicsDescription is not a function
	<<print lib.createTippyFull(setup.politicsDescription($town, "economicIdeology"), $town.economicIdeologyIST)>> instrument.js:109:45
Error: <<print>>: bad evaluation: setup.politicsDescription is not a function
	<<print lib.createTippyFull(setup.politicsDescription($town, "politicalIdeology"), $town.politicalIdeologyIC)>> instrument.js:109:45
Error: <<print>>: bad evaluation: setup.politicsDescription is not a function
	<<print lib.createTippyFull(setup.politicsDescription($town, "politicalSource"), $town.politicalSource)>> instrument.js:109:45
Error: <<listbox>>: bad evaluation: can't convert undefined to object
	<<listbox "$newBuilding">><<optionsfrom Object.keys(setup.buildingTypes)>>
	<</listbox>> instrument.js:109:45
Error: <<for>>: bad range expression: State.variables.town is undefined
	<<for _roadKey, _road range $town.roads>><<capture _roadKey, _road>>
	  \<<for _buildingKey, _obj range _road.inhabitants.buildings>><<set _building to lib.findInArray($town.buildings, "key", _buildingKey)>><<capture _buildingKey, _building>>
	    \<<if !Object.keys(_road.inhabitants.buildings).length>><<continue>><</if>>
	    \<<if !_building>><<continue>><</if>>
	    \<<if _building.objectType === 'room' || _building.parentKey>><<continue>><</if>>
	    \<<if def _previousBuilding && _previousBuilding.road === _roadKey>>
	
	      \<<print ["Also on _road.name is ", "Down from _previousBuilding.name is ", "Nearby is "].random()>>
	    \<<elseif _road.precedingText>>
	    
	    _road.precedingText <<else>>
	    
	
	    \<<print ["On ", "Along ", "Over on "].random()>> <<print lib.createTippyFull($town.roads[_roadKey].description, $town.roads[_roadKey].name)>> is 
	    \<</if>>
	    \<<if _building.needsWordNoun>>
	      \the _building.wordNoun 
	    \<</if>>
	    \<span @id='_building.name'><<include "BuildingLink">></span><<set _previousBuilding to _building>><</capture>>
	  \<</for>>
	  \<</capture>>
	\<</for>> instrument.js:109:45
undefined instrument.js:109:45
Error: macro <<notify>> does not exist
	<<notify 500000 'cookie'>> instrument.js:109:45
Error: macro <</notify>> does not exist
	<</notify>>
```

**Comment by ryceg** [comment]
Thanks for this. Can you try with a Chromium based browser to see if it's related to the machine that you're using, or the software? Does it work on any of your other devices?

**Comment by smeagolthellama** [comment]
I tried Chromium, and it works fine. (I thought I'd sent this reply as an email, but it seems it didn't work)

**Comment by ryceg** [comment]
Chromium on the same device? Okay, that's useful. 

Please do not message bugs privately. The bug tracker is public so anyone can work on fixing the issue, and people having similar issues can see it and not need to make a new thread- sending it to me would just complicate it slightly. Thanks!

**Comment by foresto** [comment]
I had a similar experience today: the Generator produced errors, and even after I allowed scripts from all domains or disabled uMatrix (my script blocker) entirely, cleared cookies and cache, and forced a full reload, the errors persisted.

However, when I enabled scripts from unpkg.com *and* restarted Firefox (instead of merely clearing cookies & cache and reloading), the Generator started working.

This is puzzling, because enabling scripts and reloading is normally enough to get even the worst web apps I've encountered to function. I've never seen one before that actually requires a browser restart. I wonder what causes this bizarre behavior.
